extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,86 @@
+import json
+from packageurl import PackageURL
+
+from conan.api.output import cli_out_write
+from conan.api.conan_api import ConanAPI
+from conan.cli.command import conan_command
+
+
+def licenses(id):
+    """
+    see https://cyclonedx.org/docs/1.4/json/#components_items_licenses
+    """
+    if id is None:
+        return []
+    else:
+        return [{"license": {
+            "id": id
+        }}]
+
+
+def package_url(node: dict) -> PackageURL:
+    """
+    Creates a PURL following https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst#conan
+    """
+    qualifiers = {
+        "prev": node["prev"]
+    }
+    if node["user"]:
+        qualifiers["user"] = node["user"]
+    if node["channel"]:
+        qualifiers["channel"] = node["channel"]
+    if "#" in node["ref"]:
+        qualifiers["rref"] = node["ref"].split("#")[1]
+    qualifiers["repository_url"] = node["remote"] if node["remote"] else "https://center.conan.io"
+    return PackageURL(
+        type="conan",
+        name=node["name"],
+        version=node["version"],
+        qualifiers=qualifiers)
+
+
+def component(n: dict) -> dict:
+    """
+    see https://cyclonedx.org/docs/1.4/json/#components
+    """
+    result = {
+        "type": "library",
+        "bom-ref": n["id"],
+        "purl": package_url(n).to_string(),
+        "licenses": licenses(n["license"]),
+        "name": n["name"],
+        "version": n["version"]
+    }
+    if n["description"]:
+        result["description"] = n["description"]
+    if n["homepage"]:
+        result["externalReferences"] = [{
+            "url": n["homepage"],
+            "type": "website"
+        }]
+    return result
+
+
+@conan_command(group="Recipe")
+def create_sbom(conan_api: ConanAPI, parser, *args):
+    """
+    creates an SBOM in CycloneDX 1.4 JSON format from a Conan graph JSON
+    """
+    parser.add_argument("graph_json", help="Path to Conan generated graph JSON output file.")
+    args = parser.parse_args(*args)
+    with open(args.graph_json, 'r') as f:
+        data = json.load(f)
```

Comment:
You are loading the graph as a basic json document. I'm wondering whether it wouldn't be more sustainable to use the conan_api to load and access the data.

This is similar in sprit that you could maybe use https://pypi.org/project/cyclonedx-python-lib/ to write the SBOM (as pointed out by @jkowalleck ).

You can of course do everything manually, but this leads to duplication and higher maintenance costs.

Maybe you can also compare it to my draft:
https://github.com/andreas-hilti/cyclonedx-conan/tree/feat/conan_v2_draft

Reply:
I changed it: I use `common_graph_args` now, so the first step (create a graph JSON) is not longer required